eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageCodecs.scala

```diff
@@ -192,9 +194,11 @@ object LightningMessageCodecs {
     ("channelId" | bytes32) ::
       ("tlvStream" | TxCompleteTlv.txCompleteTlvCodec)).as[TxComplete]
 
-  private val witnessElementCodec: Codec[ByteVector] = variableSizeBytes(uint16, bytes)
-  private val witnessStackCodec: Codec[ScriptWitness] = listOfN(uint16, witnessElementCodec).xmap(s => ScriptWitness(s.toSeq), w => w.stack.toList)
-  private val witnessesCodec: Codec[Seq[ScriptWitness]] = listOfN(uint16, witnessStackCodec).xmap(l => l.toSeq, l => l.toList)
+  private val witnessCodec: Codec[ScriptWitness] = bytes.exmap(
+    b => Attempt.fromTry(Try(ScriptWitness.read(b.toArray))),
+    w => Attempt.Successful(ScriptWitness.write(w))
```

Comment:
Actually those are not really needed, as calls to the codecs are all made within a `Try`:
https://github.com/ACINQ/eclair/blob/daf947fb600db4ed2eee6096948cf73626b475f1/eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala#L100

Reply:
Done in https://github.com/ACINQ/eclair/pull/2633/commits/2de440eb289d9cf0d253689abc6b905e3d3df7bb and rebased directly on `master` since this shouldn't conflict with `splices`.